plugins/org.eclipse.emf.ecoretools.ale.ide/src/org/eclipse/emf/ecoretools/ale/ide/env/Normalized.java

```diff
@@ -34,30 +35,30 @@
  * This decorator should be used every time the environment contains workspace-relative
  * or platform-specific paths. 
  * <p>
- * Specifically, this decorator converts <strong>{@link #getBehaviors() behaviors}' URIs</strong>
+ * Specifically, this decorator converts <strong>{@link #getBehaviorsSources() behaviors}' URIs</strong>
  * from platform-specific URIs ({@code platform:/}) to more a more generic scheme ({@code file:/}).
  * <p>
  * Below is a typical use of this decorator:
- * <pre>IAleEnvironment dslFile = new Normalized(new Dsl(path));</pre>  
+ * <pre>IAleEnvironment dslFile = new Normalized(new DslConfiguration(path));</pre>  
  */
-public class Normalized implements IAleEnvironment {
+public class Normalized extends AbstractAleEnvironment {
```

Comment:
For consistency and clarification with other implementation of AbstractAleEnvironment, I suggest that this file should be named NormalizedAleEnvironment

Reply:
As this class is a decorator, I'd like to keep its name short and avoid suffixes to ease reading:
```java
IAleEnvironment env = 
    new Normalized(
        new MadeUpAleEnvironment(
            metamodels,
            behaviors
        )
    );
```
I think that it reads better when browsing the source code (we declare a new ALE environment that is "normalized") and that it is clear that `Normalized` is an `IAleEnvironment`.

That being said, I'm not happy with its name. Currently, `Normalized` is used to replace URIs of ALE files with their absolute file path. `WithAbsoluteBehaviorPaths` may thus be a better name:
```java
IAleEnvironment env = 
    new WithAbsoluteBehaviorPaths(
        new MadeUpAleEnvironment(
            metamodels,
            behaviors
        )
    );
```
What's your opinion on this?

Reply:
Rename as `WithAbsoluteBehaviorPathsAleEnvironment`. Verbose, but at least it's understandable.